lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,829 @@
+				),
+				'name'         => array(
+					'description' => __( 'The plugin name.', 'gutenberg' ),
+					'type'        => 'string',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'plugin_uri'   => array(
+					'description' => __( 'The plugin\'s website address.', 'gutenberg' ),
+					'type'        => 'string',
+					'format'      => 'uri',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+				),
+				'author'       => array(
+					'description' => __( 'The plugin author.', 'gutenberg' ),
+					'type'        => 'object',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+				),
+				'author_uri'   => array(
+					'description' => __( 'Plugin author\'s website address.', 'gutenberg' ),
+					'type'        => 'string',
+					'format'      => 'uri',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+				),
+				'description'  => array(
+					'description' => __( 'The plugin description.', 'gutenberg' ),
+					'type'        => 'object',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+					'properties'  => array(
+						'raw'      => array(
+							'description' => __( 'The raw plugin description.', 'gutenberg' ),
+							'type'        => 'string',
+						),
+						'rendered' => array(
+							'description' => __( 'The plugin description formatted for display.', 'gutenberg' ),
+							'type'        => 'string',
+						),
+					),
+				),
+				'version'      => array(
+					'description' => __( 'The plugin version number.', 'gutenberg' ),
+					'type'        => 'string',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+				),
+				'network_only' => array(
+					'description' => __( 'Whether the plugin can only be activated network-wide.', 'gutenberg' ),
+					'type'        => 'boolean',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'requires_wp'  => array(
+					'description' => __( 'Minimum required version of WordPress.', 'gutenberg' ),
+					'type'        => 'string',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'requires_php' => array(
+					'description' => __( 'Minimum required version of PHP.', 'gutenberg' ),
+					'type'        => 'string',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit', 'embed' ),
+				),
+				'text_domain'  => array(
+					'description' => __( 'The plugin\'s text domain.', 'gutenberg' ),
+					'type'        => 'string',
+					'readonly'    => true,
+					'context'     => array( 'view', 'edit' ),
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $this->schema );
+	}
+
+	/**
+	 * Retrieves the query params for the collections.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array Query parameters for the collection.
+	 */
+	public function get_collection_params() {
+		return array(
```

Comment:
get_items should be filterable by status similar to themes endpoint - 
https://github.com/WordPress/wordpress-develop/blob/5097db06aa0d923d3c8c1ae02432db5852afb2da/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L667-L676

Reply:
You should also be able to search installed plugins.

Reply:
I really think that search, per_page and page should here as well.

Reply:
Good point, I've added `search` and `status` parameters in 3d1a7d6829.

How do you see `page`, and `per_page` working? If we try and limit the number of items processed, ie not just do an `array_slice` after the whole processing, that code path is going to get quite messy.

If we just do an `array_slice` what benefit do we get at that point? We wouldn't save any processing, just the number of items returned over the wire. And that type of pagination is very easy to implement client side, and would probably wind up as a better user experience. For reference, the plugins list table just does an `array_slice`.

I also think the chances where someone doesn't want the full list of plugins is very rare. It also seems unlikely to me that someone would have enough plugins, thousands IMO, where it would be a real issue.